apps/nouns-camp/src/app/api/simulate/route.js

```diff
@@ -0,0 +1,114 @@
+import { CHAIN_ID } from "../../../constants/env.js";
+import { resolveIdentifier } from "../../../contracts";
+import { reportError } from "../../../utils/monitoring.js";
+
+const TENDERLY_API_ENDPOINT = `https://api.tenderly.co/api/v1/account/me/project/${process.env.TENDERLY_PROJECT_SLUG}`;
+const TENDERLY_SIMULATION_OPTIONS = {
+  save: false,
+  save_if_fails: true,
+  simulation_type: "full",
+};
+
+const shareSimulation = async (simulation) => {
+  if (simulation.status) return;
+
+  try {
+    const response = await fetch(
+      `${TENDERLY_API_ENDPOINT}/simulations/${simulation.id}/share`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Accept: "application/json",
+          "X-Access-Key": process.env.TENDERLY_API_KEY,
+        },
+      },
+    );
+
+    if (!response.ok) {
+      reportError(
+        new Error(
+          `Failed to share simulation, API returned ${response.status}`,
+        ),
+      );
+    }
+  } catch (error) {
+    reportError(error);
+  }
+};
+
+export async function POST(request) {
+  const body = await request.json();
+
+  const { address: executorAddress } = resolveIdentifier("executor");
+
+  const parsedTransaction = {
+    ...body,
+    from: executorAddress,
+    estimate_gas: true,
+    network_id: CHAIN_ID,
+    ...TENDERLY_SIMULATION_OPTIONS,
+  };
+
+  const response = await fetch(`${TENDERLY_API_ENDPOINT}/simulate`, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+      "X-Access-Key": process.env.TENDERLY_API_KEY,
+    },
+    body: JSON.stringify(parsedTransaction),
+  }).catch(() => {
+    return new Response(
```

Comment:
Didn’t notice this before: looks like this one always just get picked up below and turned into a 400

Reply:
i took away the catch, think it's better to throw then